Geometry_Engine/Create/IElement2D/NewInternalElement2D.cs

```diff
@@ -0,0 +1,53 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using BH.oM.Dimensional;
+using BH.oM.Geometry;
+
+using BH.oM.Base.Attributes;
+using System.ComponentModel;
+
+
+namespace BH.Engine.Geometry
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Creates a new Element2D, appropriate to the input type. For this case the appropriate type for the PlanarSurface will be a new PlanarSurface with a null ExternalBoundary and empty list of InternalBoundaries. \n" +
+                     "Method required for any IElement2D that contians internal IElement2Ds.")]
+        [Input("surface", "PlanarSurface just used to determine the appropriate type of IElement2D to create.")]
+        [Output("opening", "The created Opening as a IElement2D.")]
+        public static IElement2D NewInternalElement2D(this PlanarSurface surface)
```

Comment:
The prefix `New` seems redundant in a method within the class `Create`.
```suggestion
        public static IElement2D InternalElement2D(this PlanarSurface surface)
```

Reply:
More importantly, I really don't like that the input input `surface` is never used, or required by the method in any way, as the only requested data is the Type of the class. I would prefer that either: 
1. The method took a `Type` as input, or that 
2. The mapping between `typeof(PlanarSurface)` and a new instance of a PlanarSurface to be used as internal element was achieved in another way, e.g. via a static property somewhere
   ```cs
    public static class InternalElementMapping
    {
	    public static Dictionary<Type, IElement2D> Mapping { get; } = new Dictionary<Type, IElement2D>()
	    {
		    {typeof(PlanarSurface), (IElement2D)new PlanarSurface(null, new System.Collections.Generic.List<ICurve>())}
	    };
    }
   ```

Reply:
I now see that we have other cases of `NewInternalElement2D` methods - e.g. in the Analytical_Engine where we use Activator.CreateInstance to create the openings. If needed for current infrastructural reasons, my comments above can be dispensed - just leaving them there for reference as I don't particularly like the solution

Reply:
For correctness, if those methods are required, I would at least add an overload for each of those methods that takes a `Type` as input, and have the other method call this new one.

Reply:
Think you answered most of it yourself but will put some context here for reference.

It is basically a method that is required for some methods operating on IElement2Ds that can have cases where an opening (internal element 2d) needs to be cut in the host IElement2D. It is then required to generate a new IElement2D of a type acceptable for the host, and that is the job of this method.

The input type is to some extent used, for type switching via as dynamic/RunExtensionMethod. Calls are made as 
`IElement2d opening =  element2d.INewInternalElement2D();`

If this method was an instance method on the class (as in standard OO) with no arguments, it might have looked slightly better than what the current implementation does, but is basically doing the same thing, but in a way matching how the oM vs Engine is working.

Could also agree that the "New" is a bit redundant, alternatively, that this method should be a query method, but remember having that discussion years ago, before we landed in this solution. Happy for it to be changed, but not in this PR, as out of scope.

Also, not to sure I like the idea of changing to relying on type. That feels like it just would be a slowdown, for no real gain?

Reply:
Have also added a link to the wiki about what methods are required by the IElement framework in the desciption to this PR. Sorry, should have done that when raising it :)